### ISSUE TITLE
False positive `dead_code` lint

### ISSUE THREAD

**Issue by msrd0** [rust-lang/rust#101582]
Source Code: https://github.com/msrd0/cargo-doc2readme/tree/a176ccc8f5d9b201b1f39599f797f33280f72c49

This shows a false-positive `dead_code` warning, both with Rust 1.63.0 from ArchLinux packages as well as with Rust 1.65.0-nightly (646e9a0b9 2022-09-02) from rustup:

```
    Checking cargo-doc2readme v0.2.0-dev (/home/msrd0/git/cargo-doc2readme)
warning: associated function `new` is never used
  --> src/preproc.rs:36:9
   |
36 |     pub fn new(read: R) -> Self {
   |            ^^^
   |
   = note: `#[warn(dead_code)]` on by default

warning: `cargo-doc2readme` (bin "cargo-doc2readme") generated 1 warning
```

When renaming the function however, the code no longer compiles, and the part that doesn't compile is also not unused. This makes me believe that the code is, in fact, not dead.

```
    Checking cargo-doc2readme v0.2.0-dev (/home/msrd0/git/cargo-doc2readme)
error[E0599]: no function or associated item named `new` found for struct `Preprocessor` in the current scope
   --> src/input.rs:138:35
    |
138 |         let mut preproc = Preprocessor::new(read);
    |                                         ^^^
    |                                         |
    |                                         function or associated item not found in `Preprocessor<_>`
    |                                         help: there is an associated function with a similar name: `new0`
    |
   ::: src/preproc.rs:22:1
    |
22  | pub struct Preprocessor<L>
    | -------------------------- function or associated item `new` not found for this struct

warning: unused import: `Read`
  --> src/input.rs:10:32
   |
10 |     io::{self, BufReader, Cursor, Read, Write},
   |                                   ^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

For more information about this error, try `rustc --explain E0599`.
warning: `cargo-doc2readme` (lib) generated 1 warning
error: could not compile `cargo-doc2readme` due to previous error; 1 warning emitted
```

**Comment by cjgillot** [comment]
You crate has 2 targets: one library at `lib.rs`, and one binary at `bin.rs`, that are compiled separately.  The reported code is used by the library target, but not by the binary target.  The compilation for the binary target correctly reports it as dead.  This is intended behaviour.

**Comment by msrd0** [comment]
Oh wow so the binary target actually compiled the same code twice. Never thought that might happen, but makes total sense now that I think about it.
